### PR TITLE
More doctests for array fun

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Comment by fredrikekre** [comment]
Since the plan is to move out `SparseArrays`, perhaps just use `Vector`s here to simplify the excision?

**Comment by kshyatt** [comment]
Rebased, addressed comment, added `# Examples` in appropriate spots, `make docs` passed locally

File: base/subarray.jl [hunk 1]
```diff
@@ -553,13 +553,34 @@ end
...
-that appear explicitly in the given `expression`, not array slicing that
-occurs in functions called by that code.
+!!! note
+    The `@views` macro only affects `array[...]` expressions
+    that appear explicitly in the given `expression`, not array slicing that
+    occurs in functions called by that code.
+
+# Examples
+```jldoctest
+julia> A = zeros(3, 3)
```
Comment (Sacha0) [inline base/subarray.jl]: Perhaps suppress?

File: base/multidimensional.jl [hunk 2]
```diff
@@ -956,6 +962,33 @@ accumulate(op, x::AbstractVector) = accumulate(op, x, 1)
 
 Cumulative operation `op` on `A` along the dimension `dim`, storing the result in `B`.
 See also [`accumulate`](@ref).
+
+# Examples
+```jldoctest
+julia> A = [1 2; 3 4]
```
Comment (Sacha0) [inline base/multidimensional.jl]: Perhaps suppress `A` and `B`?

**Review by Sacha0 — APPROVED** [review]
Thanks @kshyatt! :)

File: base/subarray.jl (hunk 1 above)
Comment (kshyatt) [reply base/subarray.jl]: I wanted to make the change p obvious, I guess. Do you think it's too long?

Comment (Sacha0) [reply base/subarray.jl]: I found the `A = zeros(3, 3)` sufficient, and all other things being equal prefer compactness, hence :).

Comment (kshyatt) [reply base/subarray.jl]: k

**Pull request merged**